app/models/post.rb

```diff
@@ -0,0 +1,12 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+  def update_posts_counter
+    update(posts_counter: comments.count)
+  end
```

Comment:
- [ ] Kindly fix this function as well. Here I can see that you are calling count on the comments instead of posts (comments.count).
```Ruby
def update_posts_counter
  author.increment!(:posts_counter)
end
```